server/get_payload.go

```diff
@@ -406,61 +448,78 @@ func convertSSZToJSON(ethConsensusVersion string, sszBytes []byte) ([]byte, erro
-				out.Version = spec.DataVersionElectra
-				out.Electra = new(eth2ApiV1Electra.SignedBlindedBeaconBlock)
-				return out.Electra.UnmarshalSSZ(in)
-			default:
-				return errInvalidForkVersion
-			}
-		} else {
+		if ethConsensusVersion == "" {
 			return types.ErrMissingEthConsensusVersion
 		}
-	case MediaTypeJSON:
-		var err error
-		electraBlock := new(eth2ApiV1Electra.SignedBlindedBeaconBlock)
-		err = json.Unmarshal(in, electraBlock)
-		if err == nil {
-			out.Version = spec.DataVersionElectra
-			out.Electra = electraBlock
-			return nil
-		}
-		denebBlock := new(eth2ApiV1Deneb.SignedBlindedBeaconBlock)
-		err = json.Unmarshal(in, denebBlock)
-		if err == nil {
+		switch ethConsensusVersion {
+		case EthConsensusVersionBellatrix:
+			out.Version = spec.DataVersionBellatrix
+			out.Bellatrix = new(eth2ApiV1Bellatrix.SignedBlindedBeaconBlock)
+			return out.Bellatrix.UnmarshalSSZ(in)
+		case EthConsensusVersionCapella:
+			out.Version = spec.DataVersionCapella
+			out.Capella = new(eth2ApiV1Capella.SignedBlindedBeaconBlock)
+			return out.Capella.UnmarshalSSZ(in)
+		case EthConsensusVersionDeneb:
 			out.Version = spec.DataVersionDeneb
-			out.Deneb = denebBlock
-			return nil
+			out.Deneb = new(eth2ApiV1Deneb.SignedBlindedBeaconBlock)
+			return out.Deneb.UnmarshalSSZ(in)
+		case EthConsensusVersionElectra:
+			out.Version = spec.DataVersionElectra
+			out.Electra = new(eth2ApiV1Electra.SignedBlindedBeaconBlock)
+			return out.Electra.UnmarshalSSZ(in)
+		case EthConsensusVersionFulu:
+			out.Version = spec.DataVersionFulu
+			out.Fulu = new(eth2ApiV1Electra.SignedBlindedBeaconBlock)
+			return out.Fulu.UnmarshalSSZ(in)
+		default:
+			return errInvalidForkVersion
 		}
-		capellaBlock := new(eth2ApiV1Capella.SignedBlindedBeaconBlock)
-		err = json.Unmarshal(in, capellaBlock)
-		if err == nil {
-			out.Version = spec.DataVersionCapella
-			out.Capella = capellaBlock
-			return nil
+	case MediaTypeJSON:
+		if ethConsensusVersion == "" {
+			return types.ErrMissingEthConsensusVersion
 		}
-		bellatrixBlock := new(eth2ApiV1Bellatrix.SignedBlindedBeaconBlock)
-		err = json.Unmarshal(in, bellatrixBlock)
-		if err == nil {
+		var err error
+		switch ethConsensusVersion {
+		case EthConsensusVersionBellatrix:
+			block := new(eth2ApiV1Bellatrix.SignedBlindedBeaconBlock)
+			if err = json.Unmarshal(in, block); err != nil {
+				return err
+			}
 			out.Version = spec.DataVersionBellatrix
-			out.Bellatrix = bellatrixBlock
-			return nil
+			out.Bellatrix = block
+		case EthConsensusVersionCapella:
+			block := new(eth2ApiV1Capella.SignedBlindedBeaconBlock)
+			if err = json.Unmarshal(in, block); err != nil {
+				return err
+			}
+			out.Version = spec.DataVersionCapella
+			out.Capella = block
+		case EthConsensusVersionDeneb:
+			block := new(eth2ApiV1Deneb.SignedBlindedBeaconBlock)
+			if err = json.Unmarshal(in, block); err != nil {
+				return err
+			}
+			out.Version = spec.DataVersionDeneb
+			out.Deneb = block
```

Comment:
This whole api is so cursed. `out` should just have a type `block any`

Reply:
Lol. I don't think making it `any` would be a wise change. It's okay to be explicit.

Reply:
agree, that'd be nice! perhaps open an issue to remind us of it later and we do a follow-up PR for this?